File: OrchidCore/src/main/java/com/eden/orchid/impl/compilers/text/TextCompiler.java
package com.eden.orchid.impl.compilers.text;

import com.eden.orchid.api.compilers.OrchidCompiler;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.Map;

@Singleton
public final class TextCompiler extends OrchidCompiler {

    @Inject
    public TextCompiler() {
        super(800);
    }

    @Override
    public String compile(String extension, String source, Map<String, Object> data) {
        return source;
    }

    @Override
    public String getOutputExtension() {
        return "html";
    }

    @Override
    public String[] getSourceExtensions() {
        return new String[]{"txt", "text"};
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/compilers/text/TextCompiler.kt
package com.eden.orchid.impl.compilers.text

import com.eden.orchid.api.compilers.OrchidCompiler
import com.eden.orchid.api.options.annotations.Archetype
import com.eden.orchid.api.options.archetypes.ConfigArchetype
import com.eden.orchid.api.resources.resource.OrchidResource
import java.io.OutputStream
import java.io.OutputStreamWriter

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
@Archetype(value = ConfigArchetype::class, key = "services.compilers.txt")
class TextCompiler
@Inject
constructor() : OrchidCompiler(800) {

    override fun compile(os: OutputStream, resource: OrchidResource?, extension: String, input: String, data: MutableMap<String, Any>?) {
        OutputStreamWriter(os).append(input).close()
    }

    override fun getOutputExtension(): String {
        return "html"
    }

    override fun getSourceExtensions(): Array<String> {
        return arrayOf("txt", "text")
    }
}
